legend-engine-xts-authentication/legend-engine-xt-authentication-connection-factory/src/main/java/org/finos/legend/connection/IdentitySpecification.java

```diff
@@ -0,0 +1,113 @@
+// Copyright 2023 Goldman Sachs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package org.finos.legend.connection;
+
+import org.eclipse.collections.api.factory.Lists;
+import org.finos.legend.engine.shared.core.identity.Credential;
+import org.pac4j.core.profile.CommonProfile;
+
+import javax.security.auth.Subject;
+import java.util.List;
+import java.util.Objects;
+
+public class IdentitySpecification
```

Comment:
How is IdentitySpecification different from Identity ?

Reply:
It's the specification that the `IdentityFactory` can use to build the identity. We can cut this, but one of the reason I introduced this is to avoid modifying the `Identity` class, I was thinking of something like `new Identity.Builder()`, but then we won't need the `IdentityFactory` at all, do you prefer it that way?